src/main/java/uk/gov/hmcts/reform/iacaseapi/infrastructure/health/ServiceHealthIndicator.java

```diff
@@ -0,0 +1,54 @@
+package uk.gov.hmcts.reform.iacaseapi.infrastructure.health;
+
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.HealthIndicator;
+import org.springframework.boot.actuate.health.Status;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+
+public class ServiceHealthIndicator implements HealthIndicator {
+
+    private String uri;
+    private String matcher;
+    private RestTemplate restTemplate;
+
+    public ServiceHealthIndicator(String uri, String matcher, RestTemplate restTemplate) {
+        this.uri = uri;
+        this.matcher = matcher;
+        this.restTemplate = restTemplate;
+    }
+
+
+    @Override
+    public Health health() {
+
+        try {
+            ResponseEntity<String> response = restTemplate
+                .getForEntity(uri, String.class);
+
+            String responseBody = response.getBody().replaceAll("\\s", "");
+
+            if (response.getStatusCode() == HttpStatus.OK
+                && responseBody.contains(matcher)
+                && !responseBody.contains("DOWN")) {
+
+                return new Health
+                    .Builder(Status.UP)
+                    .build();
+            } else {
+
+                return new Health
+                    .Builder(Status.DOWN)
+                    .build();
+            }
+        } catch (RestClientException ex) {
+
```

Comment:
Put a log entry in here to say a http exception was received including the status code / message?